google-beta/resource_compute_snapshot.go

```diff
@@ -1,220 +1,536 @@
+// ----------------------------------------------------------------------------
+//
+//     ***     AUTO GENERATED CODE    ***    AUTO GENERATED CODE     ***
+//
+// ----------------------------------------------------------------------------
+//
+//     This file is automatically generated by Magic Modules and manual
+//     changes will be clobbered when the file is regenerated.
+//
+//     Please read more about how to change this file in
+//     .github/CONTRIBUTING.md.
+//
+// ----------------------------------------------------------------------------
+
 package google
 
 import (
 	"fmt"
 	"log"
+	"reflect"
+	"strconv"
 	"time"
 
+	"github.com/hashicorp/terraform/helper/customdiff"
 	"github.com/hashicorp/terraform/helper/schema"
-	"google.golang.org/api/compute/v1"
-	"google.golang.org/api/googleapi"
+	compute "google.golang.org/api/compute/v1"
 )
 
+func customDiffComputeSnapshotSnapshotEncryptionKeys(diff *schema.ResourceDiff, meta interface{}) error {
+	// oC -> Old Convenience Value
+	// nC -> New Convenience Value
+	// oN -> Old New Field Value
+	// nN -> New New Field Value
+	oC, nC := diff.GetChange("snapshot_encryption_key_raw")
+	oN, nN := diff.GetChange("snapshot_encryption_key.0.raw_key")
+
+	if nC != "" && nN != "" {
+		return fmt.Errorf("can't use snapshot_encryption_key_raw and snapshot_encryption_key.0.raw_key at the same time." +
+			"If you're removing snapshot_encryption_key.0.raw_key, set the value to \"\" instead. This is due to limitations in terraform plan.")
```

Comment:
I'd drop the `This is due to limitations in terraform plan.` bit, or at least make it due to limitations just in Terraform. I don't know that it'll be clarifying or useful information for users.